### ISSUE TITLE
Incorrect throwing error for no-constant-condition

### ISSUE THREAD

**Issue by Beraliv** [palantir/tslint#4779]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: move from `3.1.6` to `3.4.5`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
type A = {
    a: boolean;
    b?: string;
};

type Primitive = undefined | null | string | number | boolean;

function isObject<T extends Record<string, any>>(value: T | Primitive): value is T {
    const type = typeof value;
    return Boolean(value) && (type === 'object' || type === 'function');
}

function checkA(a: A | undefined): void {
    if (a !== undefined) {
        // OK in 3.1.6
        // Error in 3.4.5: unexpected constant condition
        if (isObject(a)) {
            console.log(a);
        }
    }
}

checkA(undefined); // OK
checkA({ a: true }); // OK
checkA({ a: true, b: 'Jack' }) // OK
```

with `tslint.json` configuration:

```json
{
    "extends": ["tslint:recommended"],
    "rules": {
        "no-constant-condition": true
    }
}
```

#### Actual behavior

Error in 3.4.5: unexpected constant condition

#### Expected behavior

`isObject(a)` returns `boolean` and should not throw error
![Screenshot 2019-06-27 at 21 00 53](https://user-images.githubusercontent.com/2991847/60289261-b1a59c00-991e-11e9-890f-517576adc990.png)

**Comment by Beraliv** [comment]
It was an issue of tslint-eslint-rules, sorry
